src/narjure/memory_management/concept.clj

```diff
@@ -1,30 +1,80 @@
 (ns narjure.memory-management.concept
   (:require
+    [co.paralleluniverse.pulsar.actors
+     :refer [! spawn gen-server register! cast! Server self
+             shutdown! unregister! set-state! state whereis]]
     [narjure.actor.utils :refer [defactor]]
     [taoensso.timbre :as t])
   (:refer-clojure :exclude [promise await]))
 
-(declare concept task-req belief-req inference-req persistence-req exit)
+(defn debug [msg] (t/debug :concept msg))
 
-(defactor concept
-          "State is a map
-          {:name :budget :activation-level :belief-tab :goal-tab :task-bag :term-bag}
-          (this list may not be complete)."
-          {:task-msg            task-req
-           :belief-req-msq      belief-req
-           :inference-req-msq   inference-req
-           :persistence-req-msg persistence-req})
+(defn task-handler
+  ""
+  [from message]
+  ())
 
-(defn debug [msg] (t/debug :concept msg))
+(defn belief-request-handler
+  ""
+  [from message]
+  ())
+
+(defn inference-request-handler
+  ""
+  [from message]
+  ())
+
+(defn concept-state-handler
+  ""
+  [from message]
+  ())
+
+(defn task-budget-update-handler
+  ""
+  [from message]
+  ())
+
+(defn set-content-handler
+  "Initilise the cocnept state with the term that is the content.
+   This is sent from concept-creator on creation"
+  [from [msg content]]
+  (set-state! (assoc @state :name content ))
+  #_(debug (str "set-content-msg: " content)))
 
-(defn task-req [_ _]
-  #_(debug "process-task"))
+(defn shutdown-handler
+  "Processes :shutdown-msg and shuts down actor"
+  [from msg]
+  (unregister!)
+  (shutdown!))
 
-(defn belief-req [_ _]
-  (debug "process-belief-req"))
+(defn initialise
+  "Initialises actor: registers actor and sets actor state"
+  []
+  (set-state! {:name :name
+               :budget '(0.0 0.0)
```

Comment:
Usually there is no reason to use lists instead of vectors in such places, at least vectors help to avoid quoting.